pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -76,6 +79,15 @@ var (
 	ErrVersionNotSupported = errors.New("runtime api version is not supported")
 )
 
+// Resource kinds for which In-place pod vertical scaling is applicable
+type ResizeResourceKind string
+
+const (
+	ResizeMemoryLimit ResizeResourceKind = "memory-limit"
+	ResizeCPULimit    ResizeResourceKind = "cpu-limit"
+	ResizeCPURequest  ResizeResourceKind = "cpu-request"
+)
```

Comment:
We should use the existing type definitions and not add new ones here.

Reply:
There's no existing type for CPU limit and CPU request. This was done as we need to update the resources types separately. All of this stuff has been discussed in the past. Please go and review the discussion history. There's a lot of changes done after lot of discussions in this PR and its predecessor https://github.com/kubernetes/kubernetes/pull/92127 . It's not humanly possible to remember reasons for every single change.

Reply:
I was able to get rid of this - discovered I don't really need to break apart CPU request and limit updates entirely in order get the pod cgroup request/limit pre- post- ordering correct. Next commit will get rid of this.

Reply:
Fixed in latest commit, please review.